### ISSUE TITLE
Changes to prefix not supported

### ISSUE THREAD

**Issue by pghalliday** [jssjr/git#62]
I just ran into this issue while installing git from source on a centos Jenkins slave in order to upgrade the version. I initially ran the recipe with the default prefix of `/usr/local`. This was fine except that I realised that Jenkins jobs do not run in a real login session and as such `/usr/local/bin` is not on the path (or at least not before `/usr/bin` where the previous version of git was installed). This was confusing as when I logged into the machine `git --version` would report `2.0.0` while Jenkins jobs would get `1.7.1`.

The obvious fix was to update my recipe to override the prefix to `/usr`. However when I do this the recipe does not actually do anything as there is a `not_if` check on the git version which gets a path as if it were a login session.

I think the fix is to add the prefix to the `not_if` check to make sure the correct version is installed where it is expected. I think it may also be required to remove any old build directory too (at least so `mkdir` doesn't fail but also to clean any artifacts built with the old prefix).

Something like this in `libraries/provider_git_client_source.rb`

```ruby
          execute "Extracting and Building Git #{node['git']['version']} from Source" do
            cwd Chef::Config['file_cache_path']
            command <<-COMMAND
    (rm -rf git-#{node['git']['version']} && mkdir git-#{node['git']['version']} && tar -zxf git-#{node['git']['version']}.tar.gz -C git-#{node['git']['version']} --strip-components 1)
    (cd git-#{node['git']['version']} && make prefix=#{node['git']['prefix']} install)
  COMMAND
            not_if "#{node['git']['prefix']}/bin/git --version | grep #{node['git']['version']}"
          end
```

Without this i had to log in to the box and run the following to workaround the problem:

```
rm -rf /var/chef/cache/git-2.0.0
rm -f /usr/local/bin/git
chef-client
```

**Comment by johnbellone** [comment]
This makes sense to me. What do you think @someara ?

**Comment by someara** [comment]
This should be fixed up in 4.3.2
